trustfall_core/src/ir/indexed.rs

```diff
@@ -101,13 +99,10 @@ fn get_output_type(
 ) -> Type {
     let mut wrapped_output_type = field_type.clone();
     if component_optional_vertices.contains(&output_at) {
-        wrapped_output_type.nullable = true;
+        wrapped_output_type = wrapped_output_type.with_nullability(true);
```

Comment:
the way it stands now `Type` is immutable. I think this is one of a few cases in the entire codebase that we would have benefited from it not being immutable, although it was an easy change, and immutability is good, I think.